File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/explore/ExploreViewModel.kt
package au.com.crazybean.mobilex.kurir.modules.explore

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import au.com.crazybean.foundation.mvvm.ViewModel
import au.com.crazybean.mobilex.kurir.data.model.User
import au.com.crazybean.mobilex.kurir.repository.users.UsersRepository

class ExploreViewModel(private val repository: UsersRepository?) : ViewModel() {
    private val liveData by lazy {
        MutableLiveData<List<User>?>()
    }

    fun getContacts(query: String?): LiveData<List<User>?> {
        return liveData.also {
            repository?.getUsers(null) { users ->
                it.value = users?.filter {
                   query.isNullOrEmpty() || it.firstName?.contains(query, true) == true || it.lastName?.contains(query, true) == true
                }
            }
        }
    }
}

File: ios/Kurir/modules/explore/ExploreViewModel.swift
//
//  ExploreViewModel.swift
//  Kurir
//
//  Created by Loren on 19/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Crazybean
import Mobilex

class ExploreViewModel: ViewModel {
    private let repository: UsersRepository?
    
    init(repository: UsersRepository?) {
        self.repository = repository
    }
}

